### PR TITLE
Feature/tp listener sample

### PR DESCRIPTION
**Pull request by algattik**
Added documentation and sample for TransferProcessListener

### CODE REVIEW

File: docs/architecture/README.md [hunk 1]
```diff
@@ -22,3 +22,14 @@ Before each data transfer a contract must be offered from the provider. A consum
...
+### TransferProcessListener
+
+A consumer extension may register a listener to execute custom logic after a transfer changes state, for example, to notify an external application on the consumer side after data has been produced (i.e. the transfer moves to the completed state).
+
+```java
+transferProcessObservable = context.getService(TransferProcessObservable.class);
+transferProcessObservable.registerListener(myTransferProcessListener);
+```
+
+A sample is available at [06-file-transfer-listener](../../samples/06-file-transfer-listener).
```
Comment (marcgs) [inline docs/architecture/README.md]: We agreed with Paul to create a 4.1 sample as a variation of the 4th sample. Consider renaming.

File: samples/06-file-transfer-listener/listener/src/main/java/org/eclipse/dataspaceconnector/extensions/listener/TransferListenerExtension.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package org.eclipse.dataspaceconnector.extensions.listener;
+
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferProcessObservable;
+
+public class TransferListenerExtension implements ServiceExtension {
```
Comment (marcgs) [inline samples/06-file-transfer-listener/listener/src/main/java/org/eclipse/dataspaceconnector/extensions/listener/TransferListenerExtension.java]: Nice idea of implementing the sample as an extension on top of sample 4 👍

**Review by marcgs — APPROVED** [review]

File: docs/architecture/README.md (hunk 1 above)
Comment (algattik) [reply docs/architecture/README.md]: fixed

**Review by Izzzu — APPROVED** [review]
I went through the tutorial in the sample and all works as expected :)

**Pull request closed without merging**